R/DBConnection.R

```diff
@@ -106,6 +106,41 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' @export
+dbBreak <- structure(list(), class = "dbiAbort")
```

Comment:
Ahh shoot. @jimhester, @hadley and I have actually made a lot of progress with chunked API design that we probably need to brain dump before this PR can proceed.

I think in our original discussion we preferred return(dbBreak()) or stop(dbBreak()) because it made it clearer to the reader that normal control flow would be interrupted. But now that I think about it, it'll probably be fairly obvious in most cases anyway. 

```r
callback = function(df, i) {
  if (someTest(df)) {
    dbBreak()
  }
  # more processing
}
```

So I'm ok either way.